### PR TITLE
Fix download full results link for embedded and public questions

### LINKED ISSUE #34634: 404 when downloading results on embedded questions on sub-paths
### Describe the bug

embedded dashboards do work, but now questions are the ones that fail

related to https://github.com/metabase/metabase/issues/33244

### To Reproduce

1) run metabase on a sub path
2) create a question
3) embed it
4) try to download the results... you get a 404

see the URL on my local instance:
```https://localhost:8443/mbhttps://localhost:8443/mb/embed/question/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyZXNvdXJjZSI6eyJxdWVzdGlvbiI6NH0sInBhcmFtcyI6e30sImlhdCI6MTY5NzE0ODAxMywiX2VtYmVkZGluZ19wYXJhbXMiOnt9fQ.nSfqwLkdMgFfyQry_xjDIEpI98HyOZSwlxBh4mDo-w8.xlsx```

### Expected behavior

We should generate the right API calls

### Logs

NA

### Information about your Metabase installation

47.3 onwards

### Severity

P1

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by uladzimirdev**
Closes https://github.com/metabase/metabase/issues/34634

### Description

We used site url variable to calculate links for embedded and publicly shared questions downloading, with this PR we should respect embedding and public sharing

### How to verify

1. You need to deploy the branch on subpath:
<details><summary>Steps</summary>
<p>

1. run your local instance (frontend and backend separately), go to the question and download results (csv, json, xlsx) - make sure they show correct result
2. go to admin settings -> change metabase URL from `localhost:3000` to `localhost:9090/metabase/`
3. stop your frontend server, keep backend server running
4. build js files with `yarn build`, open `localhost:3000` and make sure js files are loaded
5. clone https://github.com/iethree/metabase-subpath-proxy: e.g. `cd .. && git clone git@github.com:iethree/metabase-subpath-proxy.git`
6. go to the  e.g. `cd ./metabase-subpath-proxy` and run `docker-compose up` (or `docker compose up`)
7. now open `localhost:9090/metabase/` - notice `/` at the end of the line, it's required

</p>
</details> 

2. enable embedding of the question 

<details><summary>Steps</summary>
<p>

0. enable embedding in the admin settings
1. go to question
2. click on sharing icon in the bottom right corner
4. click setup
5. right click on iframe -> inspect element, find iframe and copy it's url
image
6. open URL in a new tab/browser
7. click on download full results and **check that downloaded file is correct**

</p>
</details> 
3. enable public sharing of the question

<details><summary>Steps</summary>
<p>

0. enable public sharing in admin settings
1. go to question, click sharing icon in the bottom right corner
2. check checkbox and copy public link
3. open public link in a new tab
4. click on download full results and **check that downloaded file is correct**

</p>
</details> 

### Demo

https://www.loom.com/share/68ec00247a2546ce999437a6b4747c02

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Review by ranquild — CHANGES_REQUESTED** [review]
99% sure that the fix is incorrect and the issue is here https://github.com/metabase/metabase/blob/bb90185ef612f10d18d33b1015dbb8494328e009/frontend/src/metabase/query_builder/actions/downloading.ts#L136

This line doesn't take absolute URLs into account

**Comment by uladzimirdev** [comment]
@ranquild what made you think the fix is incorrect? 

if you checked code, the fix was actually in getting rid of absolute URLs as they are calculated again using `site-url`, what we do again using `api.basename`. I have an idea how to test it

**Comment by ranquild** [comment]
@uladzimirdev because you ignore the `site-url` on a sub-path, that's not correct

**Comment by uladzimirdev** [comment]
@ranquild is it possible to have `site-url` and `api.basename` different? During the previous fix I thought they should be the same

**Comment by ranquild** [comment]
@uladzimirdev yes it's possible, e.g. when the domain of the parent application is not the same as in the iframe

File: frontend/src/metabase/visualizations/components/LegacyChoropleth.jsx [hunk 1]
```diff
@@ -49,12 +49,13 @@ const LegacyChoropleth = ({
                 onMouseLeave={() => onHoverFeature(null)}
                 className={cx({ "cursor-pointer": !!onClickFeature })}
                 onClick={
-                  onClickFeature &&
-                  (e =>
-                    onClickFeature({
-                      feature: feature,
-                      event: e.nativeEvent,
-                    }))
+                  onClickFeature
```
Comment (uladzimirdev) [inline frontend/src/metabase/visualizations/components/LegacyChoropleth.jsx:52]: not related, only a warning in devtools

**Review by ranquild — APPROVED** [review]

**Review by dragonsahead — APPROVED** [review]
works for me!

**Pull request merged**